solutions/2018/01_chronal_calibration.rb

```diff
@@ -0,0 +1,24 @@
+class ChronalFrequencies
+  DEFAULT_CYCLE_MAXIMUM = 1000
+
+  def initialize(frequency_change_array)
+    @frequency_change_array = frequency_change_array
+  end
+
+  def sum_all
+    @frequency_change_array.sum
+  end
+
+  def repeat_sum(repeat_count = 1, cycle_max = DEFAULT_CYCLE_MAXIMUM)
+    results = { 0 => 1}
+    sum = 0
+
+    @frequency_change_array.cycle(cycle_max) do |f|
+      sum += f
+      return sum if results[sum] == repeat_count
+      results[sum] = results[sum].to_i + 1
+    end
+
+    raise "no repeating frequency after #{cycle_max} cycles"
```

Comment:
I would probably just return `nil` in this case - it's not exceptional behaviour, it's a none and potentially common case. Consider what Ruby's `detect` does in similiar situations:
```ruby
[1, 2, 3].detect { |e| e > 5 }
=> nil
```